### DISCUSSION TITLE
Delaying a callback sent by an actor

### DISCUSSION THREAD

**Discussion by rajansaini691** [statelyai/xstate#1777] [Q&A]
Hello again, and thanks for all of your help! I'm trying to delay a callback sent by an actor. For some reason, the `callback("evt", {delay: 1000})` syntax does not seem to work in this case. I made a repro [here](https://codesandbox.io/s/xstate-example-template-vanilla-forked-tx6k6?file=/src/index.js), showing that the delayed and non-delayed events are received at the same time. 

However, there is a non-negligible chance that my syntax is incorrect. It is based on previous discussion answers and docs. 

Any insights into why this is happening would be greatly appreciated!

**Comment by davidkpiano** [comment]
Callback actors are not machines, so when you send an event (in this example, with `callback(event)`), it is always sent immediately.

The way the `actions.send()` action creator works in machines in XState is that it creates an action object such as:

```js
{
  type: 'xstate.send',
  event: {/* ... */},
  delay: 1000
}
```

And the interpreter interprets that _action object_ (note: not an event!) to send the `actionObject.event` after the specified `actionObject.delay`.

- `actions.send(...)` - creates an _action object_ that sends an event (possibly after a delay)
- `interpreter.send(...)` - immediately sends an event; delays can't be configured
- `callback(...)` (from a callback actor) - immediately sends an event; same as above.

You'll probably want to put the event in a `setTimeout(...)` to get your desired behavior.
